portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java

```diff
@@ -33,32 +32,25 @@
 
     // Sorts the tasks in descending order based on duration.
     // Longest task comes first in the collection.
-    // Then sorts the list again by alphabetical order ascending.
+    // Then sorts the list again by task priority.
     Collections.sort(tasksList, sortByTaskDurationDescendingThenPriority);
 
     CalendarEventsGroup calendarEventsGroup =
         new CalendarEventsGroup(eventsList, workHoursStartTime, workHoursEndTime);
+
     // Create a TimeRangeGroup class for the free time ranges.
     List<TimeRange> availableTimes = calendarEventsGroup.getFreeTimeRanges();
     TimeRangeGroup availableTimesGroup = new ArrayListTimeRangeGroup(availableTimes);
 
     List<ScheduledTask> scheduledTasks = new ArrayList<ScheduledTask>();
 
     for (Task task : tasksList) {
-      Duration taskDuration = task.getDuration();
-      Optional<TimeRange> timeRangeForTask = findTimeForTask(task, availableTimesGroup);
-
-      if (!timeRangeForTask.isPresent()) {
-        continue;
-      }
-
-      // Create and append scheduled tasks.
-      TimeRange scheduledTimeRange = timeRangeForTask.get();
-      ScheduledTask scheduledTask = new ScheduledTask(task, scheduledTimeRange.start());
-      scheduledTasks.add(scheduledTask);
+      List<ScheduledTask> currentScheduledTasks = scheduleOneTask(task, availableTimesGroup);
 
-      // Delete the scheduled time range from the time range group.
-      availableTimesGroup.deleteTimeRange(scheduledTimeRange);
+      currentScheduledTasks.forEach(
```

Comment:
nit: I think you can just do `currentScheduledTasks.forEach(scheduledTasks.add);`

Reply:
Updated.